### PR TITLE
Upgrade to Molecule v2

### PR DESCRIPTION
**Pull request by jperera**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions
* Remember to set **idealista:develop** as base branch;

### Description of the Change

Upgrade to Molecule v2

### Benefits

New Molecule version :) Maybe [Scenarios](https://molecule.readthedocs.io/en/latest/configuration.html#scenario) is the most remarkable new feature.

### Possible Drawbacks

Vagrant Molecule driver is not supported any more.

### Applicable Issues

#11

### CODE REVIEW

**Review by sorobon — APPROVED** [review]

File: test-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ansible==2.4.5.0
```
Comment (dortegau) [inline test-requirements.txt:1]: This version doesn't match with version described in the Readme file (2.3.1.0) 😉

Comment (jperera) [reply test-requirements.txt:1]: I change README.md and meta/main.yml

**Comment by jnogol** [comment]
Looks like tests are failing :(

**Review by jmonterrubio — APPROVED** [review]

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[[source]]
```
Comment (jnogol) [inline Pipfile:1]: The `Pipfile` and `Pipfile.lock` should be ignored :)

File: molecule/default/verifier.yml [hunk 3]
```diff
@@ -21,7 +21,7 @@
```
Comment (jnogol) [inline molecule/default/verifier.yml:23]: IMO, this file should be updated as this one: https://github.com/idealista/java-role/blob/master/molecule/openjdk/verifier.yml

Basically, c&p and keep the `vars_files` from this file

File: molecule/default/requirements.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+- src: idealista.java-role
+  version: 3.3.0
+  name: java
```
Comment (jnogol) [inline molecule/default/requirements.yml]: Looks like naming the role as "java" is making the [test](https://circleci.com/gh/idealista/jira-role/45?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) fail:

```
extracting java to /home/circleci/jira-role/molecule/default/.molecule/roles/java

...

ERROR! the role 'java-role' was not found in /home/circleci/jira-role/molecule/default/roles:/home/circleci/jira-role/molecule/default/.molecule/roles:/home/circleci:/home/circleci/jira-role/molecule/default
```

File: molecule/default/verifier.yml (hunk 3 above)
Comment (jnogol) [reply molecule/default/verifier.yml:23]: If do so, `molecule/default/tests/jira.yml.j2` must be renamed to `test_jira.yml` (`test_default.yml`, basically any `test_SOMETHING.yml` will do the trick)

**Review by dortegau — APPROVED** [review]

File: Pipfile (hunk 2 above)
Comment (dortegau) [reply Pipfile:1]: Should be? 

[Should Pipfile.lock be committed to version control?](https://github.com/pypa/pipenv/issues/598)

I had added all this files to the .gitignore file, but I don't know if should be ignored or not using pipenv to test Ansible Roles 🙁

Comment (jnogol) [reply Pipfile:1]: Well, let's leave them for the moment and see if anything explodes :laughing:

**Review by jnogol — APPROVED** [review]

File: Pipfile (hunk 2 above)
Comment (jperera) [reply Pipfile:1]: ok

**Pull request merged**